kotlinx-coroutines-core/jvm/src/AbstractTimeSource.kt

```diff
@@ -28,12 +28,8 @@ internal inline fun mockTimeSource(source: AbstractTimeSource?) {
 }
 
 @InlineOnly
-internal inline fun currentTimeMillis(): Long =
-    timeSource?.currentTimeMillis() ?: System.currentTimeMillis()
-
-@InlineOnly
-internal actual inline fun nanoTime(): Long =
-    timeSource?.nanoTime() ?: System.nanoTime()
```

Comment:
This approach seems too invasive to me, and together with the added boxing throughout our implementation, it looks like it can lead to some observable performance degradation.

Here's an alternative proposal: `internal fun nanoTime(): Long = (TimeSource.Monotonic.markNow() - startingPoint).inWholeNanoseconds; private val startingPoint: ValueTimeMark = TimeSource.Monotonic.markNow()`. This way, `startingPoint` will get initialized once per run, and the rest of the program can be kept the way it is.

Reply:
Perfect. Please take a look.